lib/nostr.js

```diff
@@ -187,48 +179,10 @@ export function nostrZapDetails (zap) {
   return { npub, content, note }
 }
 
-async function publishNostrEvent (signedEvent, relayUrl) {
-  const timeout = 3000
-  const relay = await Relay.connect(relayUrl, { timeout })
-  try {
-    await relay.publish(signedEvent, { timeout })
-  } finally {
-    relay.close()
-  }
-}
-
-export async function crosspost (event, relays = DEFAULT_CROSSPOSTING_RELAYS) {
-  try {
-    const signedEvent = await callWithTimeout(() => window.nostr.signEvent(event), 10000)
-    if (!signedEvent) throw new Error('failed to sign event')
-
-    const promises = relays.map(r => publishNostrEvent(signedEvent, r))
-    const results = await Promise.allSettled(promises)
-    const successfulRelays = []
-    const failedRelays = []
-
-    results.forEach((result, index) => {
-      if (result.status === 'fulfilled') {
-        successfulRelays.push(relays[index])
-      } else {
-        failedRelays.push({ relay: relays[index], error: result.reason })
-      }
-    })
-
-    let noteId = null
-    if (signedEvent.kind !== 1) {
-      noteId = await nip19.naddrEncode({
-        kind: signedEvent.kind,
-        pubkey: signedEvent.pubkey,
-        identifier: signedEvent.tags[0][1]
-      })
-    } else {
-      noteId = hexToBech32(signedEvent.id, 'note')
-    }
-
-    return { successfulRelays, failedRelays, noteId }
-  } catch (error) {
-    console.error('Crosspost error:', error)
-    return { error }
+// workaround NDK url parsing issue (see https://github.com/stackernews/stacker.news/pull/1636)
+class NDKNip46SignerURLPatch extends NDKNip46Signer {
+  connectionTokenInit (connectionToken) {
+    connectionToken = 'http' + connectionToken.substring(connectionToken.indexOf('://'))
```

Comment:
Can this be anything else than `bunker://`? If not, I think this is easier to read and does the same:

```suggestion
    connectionToken = connectionToken.replace('bunker://', 'http://')
```

Reply:
@riccardobl did you miss this suggestion?